packages/data-api/src/fetchData.js

```diff
@@ -121,7 +106,7 @@ const generateBaseSqlQuery = ({
   endDate,
   aggregations,
 }) => {
-  const firstAggregationType = aggregations && aggregations[0] && aggregations[0].type;
+  const firstAggregation = AGGREGATIONS[aggregations?.[0]?.type] || AGGREGATIONS.DEFAULT;
```

Comment:
Just checking, I remember in some visualisations, we do want to perform the entityAggregation first before the input aggregations (by having a config `entityAggregationOrder: 'BEFORE' || 'AFTER'`) Will that still work in this case? Coz I think this is assuming that the input aggregation is the first one.

Maybe the worst case is just it will just do the input aggregations not on the fetchData level?

Reply:
No you're right, we won't support doing aggregation in the fetchData level for that scenario, it will be done by the aggregator instead. Really we only support quite a specific scenario (which actually happens to be rather common) where the first aggregation in the chain is one of the supported ones. I think it will be too difficult to try and support multiple or chained aggregations in the fetchData level, though it may be possible if we wish (eg. We could support [MOST_RECENT, SUM], if we wrap the most recent request with another request for the sum).

Ensuring that we get the most utility out of aggregating on the dataFetch will require us to be mindful when creating dashboards/overlays of how the data is being processed throughout the pipeline. It puts a bit more onus on us to write our configs in a way that yields the best performance.

Reply:
thanks for being patient with all my questions!